aws/table_aws_wellarchitected_notification.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/aws/aws-sdk-go-v2/service/wellarchitected"
+
+	wellarchitectedv1 "github.com/aws/aws-sdk-go/service/wellarchitected"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsWellArchitectedNotification(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_wellarchitected_notification",
+		Description: "AWS Well-Architected Notification",
+		List: &plugin.ListConfig{
+			Hydrate: listWellArchitectedNotifications,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "workload_id", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(wellarchitectedv1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "current_lens_version",
+				Description: "The current version of the lens.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensUpgradeSummary.CurrentLensVersion"),
+			},
+			{
+				Name:        "latest_lens_version",
+				Description: "The latest version of the lens.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensUpgradeSummary.LatestLensVersion"),
+			},
+			{
+				Name:        "lens_alias",
+				Description: "The alias of the lens.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensUpgradeSummary.LensAlias"),
+			},
+			{
+				Name:        "lens_arn",
+				Description: "The ARN of the lens.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensUpgradeSummary.LensArn"),
+			},
+			{
+				Name:        "notification_type",
+				Description: "The type of notification.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Type"),
+			},
+			{
+				Name:        "workload_id",
+				Description: "The ID assigned to the workload.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensUpgradeSummary.WorkloadId"),
+			},
+			{
+				Name:        "workload_name",
+				Description: "The name of the workload.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("LensUpgradeSummary.WorkloadName"),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listWellArchitectedNotifications(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := WellArchitectedClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_notification.listWellArchitectedNotifications", "client_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxLimit := int32(50)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
```

Comment:
@karanpopat I think we can remove this check since the SDK already handles it for us.